LiteCore/Support/Error.cc

```diff
@@ -379,14 +450,58 @@ namespace litecore {
     }
 #endif // LITECORE_IMPL
 
+    error error::make_error(Domain domain, int code) {
+        if(domain == POSIX && (code <= MinPosixErrorMinus1 || code >= MaxPosixErrorPlus1)) {
+            for (const codeMapping *row = &kPOSIXMapping[0]; row->err != 0; ++row) {
+                if(row->err == code) {
+                    return error(row->domain, row->code, _what(row->domain, row->code));
+                }
+            }
+
+            DebugAssert(false, "Invalid POSIX code for make_error: %d", code);
+            return error(error::UnexpectedError);
+        }
+
+        return error(domain, code, _what(domain, code));
```

Comment:
why not only return this `return error::make_error(domain, code, _what(domain, code))` (without the above checks) since its already checked in the below function?

Reply:
It needs to translate the domain and code first to get the correct error message out of `_what`

Reply:
This creates a weird dichotomy — when does one use the `error` constructor and when `make_error`? It seems like the only need for `make_error` is to convert platform to standard POSIX codes, so could we change this to `makePOSIXError` and use the constructor for everything else?